pymc3/step_methods/hmc/base_hmc.py

```diff
@@ -38,15 +42,18 @@ def __init__(self, vars=None, scaling=None, step_scale=0.25, is_cov=False,
             vars = model.cont_vars
         vars = inputvars(vars)
 
-        if scaling is None:
+        if scaling is None and potential is None:
             scaling = model.test_point
 
         if isinstance(scaling, dict):
             scaling = guess_scaling(Point(scaling, model=model), model=model, vars=vars)
 
-        n = scaling.shape[0]
+        if scaling is not None and potential is not None:
+            raise ValueError("Can not specify both potential and scaling.")
+
+        n = model.dict_to_array(model.test_point).shape[0]
         self.step_size = step_scale / (n ** 0.25)
-        self.potential = quad_potential(scaling, is_cov, as_cov=False)
+        self.potential = quad_potential(scaling or potential, is_cov, as_cov=False)
```

Comment:
I think this should be outside, `quad_potential()`, no? Also, I think better to be explicit here, i.e. :`if potential is not None: self.potential = potential`.